pkg/task/storage.go

```diff
@@ -0,0 +1,155 @@
+package task
+
+import (
+	"container/heap"
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	"github.com/syndtr/goleveldb/leveldb"
+	"github.com/syndtr/goleveldb/leveldb/opt"
+)
+
+// TaskStorage is a leveldb-backed priority queue.
+// storage is persisted before the in-memory queue to prevent inconsistency between restarts; what
+// you see in the queue is the same as what is written the database
+type TaskStorage struct {
```

Comment:
The task data model will evolve over time. It's important to incorporate a "schema version" field at the database level since the beginning.

- New testground versions may update the task schema.
- On `Load`, we should perform any [schema migration](https://en.wikipedia.org/wiki/Schema_migration) by applying incremental migration logic.

Reply:
I agree we need a `schema` version field. I'd rather we don't dive into how migrations work for now, this shouldn't be very hard as soon as we need the first migration.

Reply:
Added a version.